apis/v1beta1/gateway_types.go

```diff
@@ -148,6 +153,68 @@ type GatewaySpec struct {
 	Addresses []GatewayAddress `json:"addresses,omitempty"`
 }
 
+// GatewayInfrastructure defines infrastructure level attributes about a Gateway
+type GatewayInfrastructure struct {
+	// Routability allows the Gateway to specify the accessibility of it's addresses. Setting
+	// this property will override the default value defined by the [GatewayClass]
```

Comment:
it's new in 1.19 - it links to the type or package

https://go.dev/doc/comment

Reply:
That's very cool, good to know! Unfortunately it looks like our reference doc generation does not support this yet (https://deploy-preview-2157--kubernetes-sigs-gateway-api.netlify.app/references/spec/#gateway.networking.k8s.io%2fv1beta1.GatewayClassStatus). In general I think we try to make references in the API Spec something that users will understand, not implementations, so we use field names instead of type names. Before moving to an approach like this I think we should wait for broader community guidance to make that kind of change.

Reply:
Sounds good

Reply:
ok removed the links where it's broken. Some markdown still works in some cases in both godoc and the API ref docs. So I left those in place.